ec/ecresource/deploymentresource/schema_appsearch.go

```diff
@@ -102,3 +108,44 @@ func appsearchTopologySchema() *schema.Schema {
 		},
 	}
 }
+
+func appSearchConfig() *schema.Schema {
+	return &schema.Schema{
+		Type:             schema.TypeList,
+		Optional:         true,
+		MaxItems:         1,
+		DiffSuppressFunc: suppressMissingOptionalConfigurationBlock,
+		Description:      `Optionally define the AppSearch configuration options for the AppSearch Server`,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"secret_session_key": {
+					Type:        schema.TypeString,
+					Description: `Optionally override the secret session key within AppSearch - defaults to the previously existing secretSession`,
+					Computed:    true,
+				},
+
+				// User settings
```

Comment:
I don't think this line is necessary? the fields already start with `"user_settings_*"`